src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -274,10 +275,37 @@ object BoringUtils {
             // if drilling down, don't drill Probe types
             val bore = if (up) module.createSecretIO(purePortType) else module.createSecretIO(Flipped(purePortTypeBase))
             module.addSecretIO(bore)
-            if (isDrive) {
-              conLoc.asInstanceOf[RawModule].secretConnection(rhs, bore)
-            } else {
-              conLoc.asInstanceOf[RawModule].secretConnection(bore, rhs)
+
+            // TODO: Check for wiring non-probes not in same block, reject/diagnose.
+
+            // `module` contains the new port.
+            // `conLoc` is the module with the `rhs` value, and where we want to insert.
+            require(conLoc == module || Some(conLoc) == module._parent)
```

Comment:
If this is truly a `require` (implying that bad input from the user can trip it), then it must have an error message for the user to tell them what they did wrong.

Reply:
Done! It's internal but message doesn't hurt anyway 👍 .